dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/LuceneQueryDateTimeFormatter.java

```diff
@@ -0,0 +1,255 @@
+                clauses.set(clauses.size() - 1, clauses.get(clauses.size() - 1) + " " + token);
+            } else {
+                clauses.add(token);
+            }
+        }
+
+        //DOTCMS - 4127
+        final List<Field> dateFields = new ArrayList<Field>();
+        String tempStructureVarName;
+        Structure tempStructure;
+
+        for (final String clause: clauses) {
+
+            // getting structure names from query
+            if(clause.indexOf('.') >= 0 && clause.indexOf('.') < clause.indexOf(':')){
+
+                tempStructureVarName = clause.substring(0, clause.indexOf('.'));
+                tempStructure = CacheLocator.getContentTypeCache().getStructureByVelocityVarName(tempStructureVarName);
+
+                if (UtilMethods.isSet(tempStructure) && UtilMethods.isSet(tempStructure.getVelocityVarName())) {
+                    final List<Field> tempStructureFields = new ArrayList<>(FieldsCache
+                            .getFieldsByStructureVariableName(
+                                    tempStructure.getVelocityVarName()));
+
+                    for (int pos = 0; pos < tempStructureFields.size(); ) {
+
+                        if (tempStructureFields.get(pos).getFieldType()
+                                .equals(Field.FieldType.DATE_TIME.toString()) ||
+                                tempStructureFields.get(pos).getFieldType()
+                                        .equals(Field.FieldType.DATE.toString()) ||
+                                tempStructureFields.get(pos).getFieldType()
+                                        .equals(Field.FieldType.TIME.toString())) {
+                            ++pos;
+                        } else {
+                            tempStructureFields.remove(pos);
+                        }
+
+                    }
+
+                    dateFields.addAll(tempStructureFields);
+                }
+            }
+        }
+
+        String replace;
+        List<RegExMatch> matches;
+        String structureVarName;
+        for (final String clause: clauses) {
+            for (final Field field: dateFields) {
+
+                structureVarName = CacheLocator.getContentTypeCache()
+                        .getStructureByInode(field.getStructureInode()).getVelocityVarName()
+                        .toLowerCase();
+
+                if (clause.startsWith(structureVarName + "." + field.getVelocityVarName().toLowerCase() + ":") || clause.startsWith("moddate:")) {
+                    replace = new String(clause);
+                    if (field.getFieldType().equals(Field.FieldType.DATE_TIME.toString()) || clause.startsWith("moddate:")) {
+                        matches = RegEX.find(replace, "\\[(\\d{1,2}/\\d{1,2}/\\d{4}) TO ");
+                        for (final RegExMatch regExMatch : matches) {
+                            replace = replace.replace("[" + regExMatch.getGroups().get(0).getMatch() + " TO ", "["
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal " TO " appears 9 times in this file; the first occurrence is on line 165](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4545506)